### PR TITLE
Convert AWS AutoScalingGroup to awslabs/aws-sdk-go

### PR DESCRIPTION
**Pull request by catsby**
This PR converts AWS AutoScalingGroup to use `awslabs/aws-sdk-go`.

Because of light coupling with Launch Configurations, the test file keeps `mitchellh/goamz` as the ASG library for now, until I can convert the Launch Configurations resource.

### CODE REVIEW

File: builtin/providers/aws/resource_aws_autoscaling_group.go [hunk 1]
```diff
@@ -121,32 +123,27 @@ func resourceAwsAutoscalingGroup() *schema.Resource {
...
-	autoScalingGroupOpts.SetMinSize = true
-	autoScalingGroupOpts.SetMaxSize = true
-	autoScalingGroupOpts.AvailZone = expandStringList(
+	autoscalingconn := meta.(*AWSClient).awsAutoScalingconn
+
+	var autoScalingGroupOpts autoscaling.CreateAutoScalingGroupType
+	autoScalingGroupOpts.AutoScalingGroupName = aws.String(d.Get("name").(string))
+	autoScalingGroupOpts.HealthCheckType = aws.String(d.Get("health_check_type").(string))
+	autoScalingGroupOpts.LaunchConfigurationName = aws.String(d.Get("launch_configuration").(string))
+	autoScalingGroupOpts.MinSize = aws.Integer(d.Get("min_size").(int))
```
Comment (pearkes) [inline builtin/providers/aws/resource_aws_autoscaling_group.go]: I take it `aws.Integer` handles the default case?

File: builtin/providers/aws/resource_aws_autoscaling_group.go [hunk 2]
```diff
@@ -155,8 +152,10 @@ func resourceAwsAutoscalingGroupCreate(d *schema.ResourceData, meta interface{})
 	}
 
 	if v, ok := d.GetOk("vpc_zone_identifier"); ok && v.(*schema.Set).Len() > 0 {
-		autoScalingGroupOpts.VPCZoneIdentifier = expandStringList(
-			v.(*schema.Set).List())
+		exp := expandStringList(v.(*schema.Set).List())
+		log.Printf("\n\nexp:\n\t%#v\n\n", exp)
```
Comment (pearkes) [inline builtin/providers/aws/resource_aws_autoscaling_group.go]: Is this debugging?

File: builtin/providers/aws/resource_aws_autoscaling_group.go [hunk 3]
```diff
@@ -283,16 +278,16 @@ func resourceAwsAutoscalingGroupDelete(d *schema.ResourceData, meta interface{})
...
-		Names: []string{d.Id()},
+	describeOpts := autoscaling.AutoScalingGroupNamesType{
+		AutoScalingGroupNames: []string{d.Id()},
 	}
 
 	log.Printf("[DEBUG] AutoScaling Group describe configuration: %#v", describeOpts)
 	describeGroups, err := autoscalingconn.DescribeAutoScalingGroups(&describeOpts)
 	if err != nil {
-		autoscalingerr, ok := err.(*autoscaling.Error)
+		autoscalingerr, ok := err.(aws.APIError)
```
Comment (pearkes) [inline builtin/providers/aws/resource_aws_autoscaling_group.go]: Thinking out loud here, but `.Code` should stay the same, right?

Comment (mitchellh) [inline builtin/providers/aws/resource_aws_autoscaling_group.go]: Yeah, same code.

File: builtin/providers/aws/resource_aws_autoscaling_group.go (hunk 1 above)
Comment (mitchellh) [inline builtin/providers/aws/resource_aws_autoscaling_group.go]: min_size and max_size are required, and `Get` will always return a zero value if not set ("0"), so this should be fine.

File: builtin/providers/aws/resource_aws_autoscaling_group.go [hunk 4]
```diff
@@ -194,41 +193,38 @@ func resourceAwsAutoscalingGroupRead(d *schema.ResourceData, meta interface{}) e
 	d.Set("load_balancers", g.LoadBalancerNames)
 	d.Set("min_size", g.MinSize)
 	d.Set("max_size", g.MaxSize)
-	d.Set("name", g.Name)
-	d.Set("vpc_zone_identifier", strings.Split(g.VPCZoneIdentifier, ","))
+	d.Set("name", g.AutoScalingGroupName)
```
Comment (mitchellh) [inline builtin/providers/aws/resource_aws_autoscaling_group.go]: Does this need a pointer dereference,too?

**Comment by mitchellh** [comment]
LGTM, pending some basic questions.

File: builtin/providers/aws/resource_aws_autoscaling_group_test.go [hunk 5]
```diff
@@ -262,7 +264,7 @@ resource "aws_launch_configuration" "foobar" {
 
 resource "aws_autoscaling_group" "bar" {
   availability_zones = ["us-west-2a"]
-  name = "foobar3-terraform-test"
+  name = "foobar3-terraform-test-%d"
```
Comment (pearkes) [inline builtin/providers/aws/resource_aws_autoscaling_group_test.go]: Wouldn't this break assertions for `foobar3-terraform-test` above?

File: builtin/providers/aws/resource_aws_autoscaling_group.go (hunk 1 above)
Comment (pearkes) [inline builtin/providers/aws/resource_aws_autoscaling_group.go]: @mitchellh Oh, nice. Fancy

File: builtin/providers/aws/resource_aws_autoscaling_group.go (hunk 2 above)
Comment (catsby) [inline builtin/providers/aws/resource_aws_autoscaling_group.go]: lol yes

File: builtin/providers/aws/resource_aws_autoscaling_group_test.go (hunk 5 above)
Comment (catsby) [inline builtin/providers/aws/resource_aws_autoscaling_group_test.go]: Not sure I follow.. this ASG is attached to `${aws_elb.bar.name}` and `${aws_launch_configuration.foobar.name}`, neither of which reference this ASG. Am I missing a connection between them?

**Comment by catsby** [comment]
@mitchellh @pearkes updated from feedback. I'm a bit concerned that I was not dereferencing those strings before and the tests were passing. Know a way to check the state there? To make sure things are stored correctly? o.O

File: builtin/providers/aws/resource_aws_autoscaling_group_test.go (hunk 5 above)
Comment (mitchellh) [inline builtin/providers/aws/resource_aws_autoscaling_group_test.go]: Ah, there is a test step above that tests this name directly. That is what @pearkes is talking about. Do the ASG acceptance tests pass?

**Comment by pearkes** [comment]
@catsby Yea, the tests should have some stuff in there for how to check the state...i.e [this](https://github.com/hashicorp/terraform/blob/master/builtin/providers/aws/resource_aws_autoscaling_group_test.go#L34-L36)?

**Pull request merged**